full-service/src/validator_ledger_sync.rs

```diff
@@ -145,33 +143,33 @@ impl ValidatorLedgerSyncThread {
                 }
             };
 
-        let blocks_and_contents: Vec<(Block, BlockContents)> = blocks_data
-            .into_iter()
-            .map(|block_data| (block_data.block().clone(), block_data.contents().clone()))
-            .collect();
-
-        mc_ledger_sync::identify_safe_blocks(ledger_db, &blocks_and_contents, logger)
+        mc_ledger_sync::identify_safe_blocks(ledger_db, &blocks_data, logger)
     }
 
-    fn append_safe_blocks(
-        ledger_db: &mut LedgerDB,
-        blocks_and_contents: &[(Block, BlockContents)],
-        logger: &Logger,
-    ) {
+    fn append_safe_blocks(ledger_db: &mut LedgerDB, block_data: &[BlockData], logger: &Logger) {
         log::info!(
             logger,
             "Appending {} blocks to ledger, which currently has {} blocks",
-            blocks_and_contents.len(),
+            block_data.len(),
             ledger_db
                 .num_blocks()
                 .expect("failed getting number of blocks"),
         );
 
-        for (block, contents) in blocks_and_contents {
+        for block_data in block_data {
             ledger_db
-                .append_block(block, contents, None)
+                .append_block(
+                    block_data.block(),
+                    block_data.contents(),
+                    block_data.signature(),
```

Comment:
It turns out we cannot safely append the signature from upstream. I added custom handling in https://github.com/mobilecoinfoundation/mobilecoin/pull/2216, but it might suffice to just pass None here.
```suggestion
                    None,
```